src/modules/kvs/lookup.c

```diff
@@ -57,6 +57,11 @@ typedef struct {
     zlist_t *pathcomps;
 } walk_level_t;
 
+typedef struct {
```

Comment:
I am probably not understanding the big picture, but why do we need the zlist of these structs when we have the json_t ref itself, which contains a JSON array of hash keys into the cache?

Reply:
I think you're right, we don't.  I was sort of mimicking what was done in the internal commit API.

Reply:
doh!  changing up the code right now, I realize there was a reason.   We can't assume every blobref in the valref treeobj is missing from the cache.  Some could be missing while others aren't.

It perhaps wouldn't have much performance impact to pass all references back to the caller, b/c the main kvs module would already recognize that some of those missing blobrefs are already in the cache.  It'd simply be an API style change to say I'm passing back a list of references, atleast 1 of which is missing.  Instead of all of them are missing.

I'll have to ponder this.  As the change from the list to just using the valref object is much cleaner.

Reply:
duh, thought of an obvious way to fix